Jenkinsfile

```
@@ -850,93 +857,88 @@ def isChangedXPackCode(patterns) {
 
   // Heartbeat depends on libbeat only.
   // The Heartbeat x-pack build contains all functionality from OSS Heartbeat.
-  env.BUILD_HEARTBEAT = isChangedOSSCode(["^heartbeat/.*"])
-  env.BUILD_HEARTBEAT_XPACK = env.BUILD_HEARTBEAT || isChangedXPackCode([
-    "^x-pack/heartbeat/.*",
-  ])
-
+  env.BUILD_HEARTBEAT = isChangedOSSCode(getVendorPatterns('heartbeat'))
+  env.BUILD_HEARTBEAT_XPACK = isChangedXPackCode(getVendorPatterns('x-pack/heartbeat'))
 
   // Journalbeat depends on libbeat only.
   // The Journalbeat x-pack build contains all functionality from OSS Journalbeat.
-  env.BUILD_JOURNALBEAT = isChangedOSSCode(["^journalbeat/.*"])
-  env.BUILD_JOURNALBEAT_XPACK = env.BUILD_JOURNALBEAT || isChangedXPackCode([
-    "^x-pack/journalbeat/.*",
-  ])
+  env.BUILD_JOURNALBEAT = isChangedOSSCode(getVendorPatterns('journalbeat'))
+  env.BUILD_JOURNALBEAT_XPACK = isChangedXPackCode(getVendorPatterns('x-pack/journalbeat'))
 
   // Packetbeat depends on libbeat only.
   // The Packetbeat x-pack build contains all functionality from OSS Packetbeat.
-  env.BUILD_PACKETBEAT = isChangedOSSCode(["^packetbeat/.*"])
-  env.BUILD_PACKETBEAT_XPACK = env.BUILD_PACKETBEAT || isChangedXPackCode([
-    "^x-pack/packetbeat/.*",
-  ])
+  env.BUILD_PACKETBEAT = isChangedOSSCode(getVendorPatterns('packetbeat'))
+  env.BUILD_PACKETBEAT_XPACK = isChangedXPackCode(getVendorPatterns('x-pack/packetbeat'))
 
   // Winlogbeat depends on libbeat only.
   // The Winlogbeat x-pack build contains all functionality from OSS Winlogbeat.
-  env.BUILD_WINLOGBEAT = isChangedOSSCode(["^winlogbeat/.*"])
-  env.BUILD_WINLOGBEAT_XPACK = env.BUILD_WINLOGBEAT || isChangedXPackCode([
-    "^x-pack/winlogbeat/.*",
-  ])
+  env.BUILD_WINLOGBEAT = isChangedOSSCode(getVendorPatterns('winlogbeat'))
+  env.BUILD_WINLOGBEAT_XPACK = isChangedXPackCode(getVendorPatterns('x-pack/winlogbeat'))
 
   // Elastic-agent is a self-contained product, that depends on libbeat only.
   // The agent acts as a supervisor for other Beats like Filebeat or Metricbeat.
   // The agent is available as x-pack build only.
-  env.BUILD_ELASTIC_AGENT_XPACK = isChangedXPackCode([
-    "^x-pack/elastic-agent/.*",
-  ])
+  env.BUILD_ELASTIC_AGENT_XPACK = isChangedXPackCode(getVendorPatterns('x-pack/elastic-agent'))
 
   // The Kubernetes test use Filebeat and Metricbeat, but only need to be run
   // if the deployment scripts have been updated. No Beats specific testing is
   // involved.
   env.BUILD_KUBERNETES = isChanged(["^deploy/kubernetes/*"])
 
-  env.BUILD_GENERATOR = isChangedOSSCode(["^generator/.*"])
+  env.BUILD_GENERATOR = isChangedOSSCode(getVendorPatterns('generator'))
+}
 
-  env.GO_VERSION = readFile(".go-version").trim()
+/**
+  This method grab the dependencies of a Go module and transform them on regexp
+*/
+def getVendorPatterns(beatName){
+  def output = ""
+  docker.image("golang:${GO_VERSION}").inside{
```

NIT: delegate the docker pull with a retry/sleep to minimise the environmental issues when pulling from internet.

```suggestion
  def dockerImage = "golang:${GO_VERSION}"
  retry(2) {
    sleep 2
    sh "docker pull ${dockerImage}"
  }
  docker.image("${dockerImage}").inside{
```

---

sounds like a new step for the library

---

this image is in the packer cache so, in theory, it should not take it from internet